### PR TITLE
Update to work with recent (1.8) version Parsedown

### PR DESCRIPTION
**Pull request by ms502040**

### CODE REVIEW

**Review by BenjaminHoegh — APPROVED** [review]

**Comment by BenjaminHoegh** [comment]
@ms502040 Can you please fix the issue with ul li so its get listed instead of inline

**Comment by ms502040** [comment]
@BenjaminHoegh I dont know if you mean this change. If not, can you put screenshot what you mean.

**Comment by BenjaminHoegh** [comment]
**This is how it should be:**
skaermbillede 2018-06-08 kl 17 32 44

**This is what you get:**
skaermbillede 2018-06-08 kl 17 36 49

**Comment by ms502040** [comment]
Ok I updated to produce html code as github. see [README](https://github.com/ms502040/parsedown-checkbox/tree/master/test) in test directory. 
You can update your readme and add github like style:
~~~~css
.task-list-item {
     list-style-type: none;
}
.task-list-item-checkbox {
     margin: 0 0.2em 0.25em -1.6em;
     vertical-align: middle;
}
~~~~

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 phpunit.xml
 /vendor/
-
+/.idea/
```
Comment (leblanc-simon) [inline .gitignore:3]: Please, don't add specific editor rules in .gitignore (else, for each PR, we have a new line in .gitignore)

File: ParsedownCheckbox.php [hunk 2]
```diff
@@ -6,51 +6,76 @@
...
+
+    foreach ($block['element']['elements'] as &$element) {
+      if (!isset($element['handler']['argument'][0])) {
+        continue;
+      }
+      $begin_line = substr(trim($element['handler']['argument'][0]), 0, 4);
+      $re = '/.*(\s{2,})$/';
+      if ('[ ] ' === $begin_line) {
+        if (preg_match_all($re, $element['handler']['argument'][0], $matches, PREG_SET_ORDER, 0) > 0) {
+          $element['handler']['argument'][0] = trim($element['handler']['argument'][0]) . '<br>';
```
Comment (leblanc-simon) [inline ParsedownCheckbox.php:63]: I don't understand this line. If the line finished by 2 or more space, you add a `<br>`. Why ?

File: ParsedownCheckbox.php [hunk 3]
```diff
@@ -6,51 +6,76 @@
...
         }
+        $element['handler']['argument'][0] = '<input type="checkbox" class="task-list-item-checkbox" disabled /> ' . substr($element['handler']['argument'][0], 4);
+        $element['attributes'] = array('class' => 'task-list-item');
```
Comment (leblanc-simon) [inline ParsedownCheckbox.php:66]: In the master version, we have a CSS class to know if the line is closed or not. With that, we can add CSS rule (like `text-decoration: line-through;`). Could you restore this functionality.

**Review by leblanc-simon — CHANGES_REQUESTED** [review]
Sorry for the delay, and thanks you very much for your PR!

Could you adapt the tests with your PR, please.

File: ParsedownCheckbox.php (hunk 3 above)
Comment (ms502040) [reply ParsedownCheckbox.php:66]: this is exactly as in github

File: ParsedownCheckbox.php (hunk 2 above)
Comment (ms502040) [reply ParsedownCheckbox.php:63]: if I want this:
- [ ] aaaa   
next line
- [ ] bbb

after aaaa 2 more space

**Comment by leblanc-simon** [comment]
The beta version bdf632d7289ac0d67a288d9191a7a4a95c902fac is now compatible with last version (beta) of ParsedownExtra

**Pull request closed without merging**